### PR TITLE
Use servicePatch methods from cloud-provider repo in service-controller

### LINKED ISSUE #87447: Use service patch methods from cloud-provider repo
PR #87217 added service patch methods to k8s.io/cloud-provider. 
Once there is a release with that PR included, the code in service-controller and legacy-cloud-providers/gce can be updated to use those.

/area platform/gce
/assign

### PR DESCRIPTION
**Pull request by boddumanohar**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Fixes #87447

** Explanation **
PR #87217 added service patch methods to k8s.io/cloud-provider. These changes have been released in 1.18, So the code in service-controller can be updated to use this new helper function

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```
cc: @prameshj

### CODE REVIEW

File: pkg/controller/service/controller.go [hunk 1]
```diff
@@ -804,8 +804,7 @@ func (s *Controller) addFinalizer(service *v1.Service) error {
 	updated.ObjectMeta.Finalizers = append(updated.ObjectMeta.Finalizers, servicehelper.LoadBalancerCleanupFinalizer)
 
 	klog.V(2).Infof("Adding finalizer to service %s/%s", updated.Namespace, updated.Name)
-	// TODO(87447) use PatchService from k8s.io/cloud-provider/service/helpers
-	_, err := patch(s.kubeClient.CoreV1(), service, updated)
```
Comment (andrewsykim) [inline pkg/controller/service/controller.go]: Looks like we can delete the `patch.go` and `patch_test.go` now

**Review by andrewsykim — COMMENTED** [review]
/ok-to-test

File: pkg/controller/service/controller.go (hunk 1 above)
Comment (boddumanohar) [reply pkg/controller/service/controller.go]: Yes. Shall I do it as a part of this PR or shall I create a seperate PR to delete patch{_test}.go files?

Comment (andrewsykim) [reply pkg/controller/service/controller.go]: This PR please

**Comment by boddumanohar** [comment]
```
• Failure [49.595 seconds]
[sig-storage] Downward API volume
test/e2e/common/downwardapi_volume.go:37
  should provide node allocatable (cpu) as default cpu limit if the limit is not set [NodeConformance] [Conformance] [It]
  test/e2e/framework/framework.go:703
skipped 716 lines unfold_more
[Fail] [sig-storage] Downward API volume [It] should provide node allocatable (cpu) as default cpu limit if the limit is not set [NodeConformance] [Conformance] 
test/e2e/framework/util.go:798
Ran 573 of 4847 Specs in 1129.335 seconds
FAIL! -- 572 Passed | 1 Failed | 0 Pending | 4274 Skipped
```
It looks like error is not related: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/88516/pull-kubernetes-e2e-kind-ipv6/1232315459622146048

**Comment by andrewsykim** [comment]
The e2e seems like a flake but bazel needs updating.
```

  | 25s
-- | --
diff -Naupr -x _output /home/prow/go/src/k8s.io/kubernetes/pkg/controller/service/BUILD /tmp/verify-bazel.Z5oPMi/go/src/k8s.io/kubernetes/pkg/controller/service/BUILD --- /home/prow/go/src/k8s.io/kubernetes/pkg/controller/service/BUILD	2020-02-25 14:46:33.171982085 +0000 +++ /tmp/verify-bazel.Z5oPMi/go/src/k8s.io/kubernetes/pkg/controller/service/BUILD	2020-02-25 14:46:53.337758113 +0000 @@ -11,12 +11,9 @@ go_library(      deps = [          "//staging/src/k8s.io/api/core/v1:go_default_library",          "//staging/src/k8s.io/apimachinery/pkg/api/errors:go_default_library", -        "//staging/src/k8s.io/apimachinery/pkg/apis/meta/v1:go_default_library",          "//staging/src/k8s.io/apimachinery/pkg/labels:go_default_library", -        "//staging/src/k8s.io/apimachinery/pkg/types:go_default_library",          "//staging/src/k8s.io/apimachinery/pkg/util/runtime:go_default_library",          "//staging/src/k8s.io/apimachinery/pkg/util/sets:go_default_library", -        "//staging/src/k8s.io/apimachinery/pkg/util/strategicpatch:go_default_library",          "//staging/src/k8s.io/apimachinery/pkg/util/wait:go_default_library",          "//staging/src/k8s.io/apiserver/pkg/util/feature:go_default_library",          "//staging/src/k8s.io/client-go/informers/core/v1:go_default_library", @@ -36,9 +33,7 @@ go_library(    go_test(      name = "go_default_test", -    srcs = [ -        "controller_test.go", -    ], +    srcs = ["controller_test.go"],      embed = [":go_default_library"],      deps = [          "//pkg/controller:go_default_library",  Run ./hack/update-bazel.sh

```

Run `./hack/update-bazel.sh` to fix

**Comment by prameshj** [comment]
Please fix the bazel test by running ./hack/update-bazel.sh

other than that, the change looks good to me. Thanks for picking this up!

**Comment by boddumanohar** [comment]
I've ran the `./hack/update-bazel.sh` script. Even then the build is still failing.

**Review by andrewsykim — COMMENTED** [review]
/approve
/lgtm
/retest

Thanks @boddumanohar!

**Comment by boddumanohar** [comment]
@andrewsykim thanks for reviewing this.

**Pull request merged**